Sources/web3swift/Web3/Web3+GasOracle.swift

```diff
@@ -91,12 +92,12 @@ extension Web3 {
 
             /// We're explicitly checking that feeHistory is not nil before force unwrapping it.
             // swiftlint: disable force_unwrapping
-            guard feeHistory == nil, forceDropCache, feeHistory!.timestamp.distance(to: Date()) > cacheTimeout else { return feeHistory! }
-
-            feeHistory = try await eth.feeHistory(blockCount: blockCount, block: block, percentiles: percentiles)
+            guard let feeHistory = feeHistory, !forceDropCache, feeHistory.timestamp.distance(to: Date()) < cacheTimeout else {
+                return try await eth.feeHistory(blockCount: blockCount, block: block, percentiles: percentiles)
```

Comment:
Before returning result of `eth.feeHistory` it should be saved into `feeHistory` variable.